lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -279,101 +315,80 @@ def get_tls_file_path_in_charm(self, file: File) -> str:
-        """Get the vault CA certificate secret from the peer relation.
+    def _get_ca_certificate_secret(self) -> Tuple[str, str]:
+        """Get the vault CA certificate secret.
 
         Returns:
             Tuple[Optional[str], Optional[str]]: The CA private key and certificate
         """
         try:
-            peer_relation = self.charm.model.get_relation(self.peer_relation)
-            juju_secret_id = peer_relation.data[peer_relation.app].get(
-                CA_CERTIFICATE_JUJU_SECRET_KEY
-            )
-            juju_secret = self.charm.model.get_secret(id=juju_secret_id)
+            juju_secret = self.charm.model.get_secret(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
             content = juju_secret.get_content()
             return content["privatekey"], content["certificate"]
         except (TypeError, SecretNotFoundError, AttributeError):
             raise PeerSecretError(secret_name=CA_CERTIFICATE_JUJU_SECRET_KEY)
 
-    def _set_ca_certificate_secret_in_peer_relation(
+    def _set_ca_certificate_secret(
         self,
         private_key: str,
         certificate: str,
     ) -> None:
-        """Set the vault CA certificate secret in the peer relation.
+        """Set the vault CA certificate secret.
 
         Args:
             private_key: Private key
             certificate: certificate
         """
-        if not self.charm._is_peer_relation_created():
-            raise RuntimeError("Peer relation not created")
         juju_secret_content = {
             "privatekey": private_key,
             "certificate": certificate,
         }
-        juju_secret = self.charm.app.add_secret(
-            juju_secret_content, label=CA_CERTIFICATE_JUJU_SECRET_LABEL
-        )
-        peer_relation = self.charm.model.get_relation(self.peer_relation)
-        peer_relation.data[self.charm.app].update({CA_CERTIFICATE_JUJU_SECRET_KEY: juju_secret.id})
-        tlslogger.debug("Vault CA certificate secret set in peer relation")
+        self.charm.app.add_secret(juju_secret_content, label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+        tls_logger.debug("Vault CA certificate secret set")
 
-    def _remove_ca_certificate_secret_in_peer_relation(self) -> None:
-        """Removes the vault CA certificate secret in the peer relation.
-
-        Only the key from the relation is removed. The actual juju secret is unmodified.
-        """
-        if not self.charm._is_peer_relation_created():
-            raise RuntimeError("Peer relation not created")
-        peer_relation = self.model.get_relation(self.peer_relation)
-        if peer_relation:
-            peer_relation.data[self.charm.app].pop(CA_CERTIFICATE_JUJU_SECRET_KEY)
-            tlslogger.debug("Vault CA certificate secret removed from peer relation")
-
-    def ca_certificate_set_in_peer_relation(self) -> bool:
-        """Returns whether CA certificate is stored in peer relation data."""
+    def ca_certificate_secret_set(self) -> bool:
```

Comment:
We already use the `set` verb when setting it
```suggestion
    def ca_certificate_secret_exists(self) -> bool:
```

Reply:
Done